modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientConnectionContext.java

```diff
@@ -85,6 +99,15 @@
     /** Cursor counter. */
     private final AtomicLong curCnt = new AtomicLong();
 
+    /** Tx id. */
+    private final AtomicInteger txIdSeq = new AtomicInteger();
```

Comment:
Let's make it `AtomicLong` as above

Reply:
But `txId` is int by the protocol, why do we need `AtomicLong` here?

Reply:
My bad, you are right.

Reply:
By the way, why do we use `int` and not `long` in protocol?

Reply:
I think `int` is more than enough here. Transaction ids can be reused and It's practically impossible that one transaction is still active and after 4*10^9 transactions started by the same connection new one with the same id started.